File: src/dotnet/ToteChallenge/Domain/CommandHandler.cs
namespace ToteChallenge.Domain
{
    public class CommandHandler
    {
        private readonly ICommandParser _commandParser;
        private readonly Tote _context;

        public CommandHandler(ICommandParser commandParser, Tote context)
        {
            _commandParser = commandParser;
            _context = context;
        }

        public void Execute(string input)
        {
            var command = _commandParser.Parse(input);

            command.Execute(_context);
        }
    }
}

File: src/node/app/commandHandler.js
'use strict'

module.exports = class CommandHandler {
    constructor(parser, tote){
        this.parser = parser;
        this.tote = tote;
    }

    handle(input) {
        var command = this.parser.parse(input);
        command.execute(this.tote);
    }
}